### PR TITLE
datasource/alicloud_ram_roles: Fixed user flow control

### PR DESCRIPTION
**Pull request by MrWolong**

### CODE REVIEW

File: alicloud/data_source_alicloud_ram_roles.go [hunk 1]
```diff
@@ -116,12 +118,25 @@ func dataSourceAlicloudRamRolesRead(d *schema.ResourceData, meta interface{}) er
...
 	for {
-		raw, err := client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
-			return ramClient.ListRoles(request)
+		wait := incrementalWait(3*time.Second, 3*time.Second)
+		err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+			raw, err = client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
+				return ramClient.ListRoles(request)
+			})
+			if err != nil {
+				if NeedRetry(err) || IsExpectedErrors(err, []string{"your_error_code"}) {
```
Comment (Wanghx0991) [inline alicloud/data_source_alicloud_ram_roles.go:133]: 流控的错误码没有加啊这里

File: alicloud/resource_alicloud_snat_entry.go [hunk 2]
```diff
@@ -49,12 +49,14 @@ func resourceAlicloudSnatEntry() *schema.Resource {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ForceNew:      true,
+				Computed:      true,
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_snat_entry.go:52]: 为啥这两个字段加Computed?

File: alicloud/data_source_alicloud_ram_roles.go (hunk 1 above)
Comment (MrWolong) [reply alicloud/data_source_alicloud_ram_roles.go]: 已修复

File: alicloud/resource_alicloud_snat_entry.go (hunk 2 above)
Comment (MrWolong) [reply alicloud/resource_alicloud_snat_entry.go]: 已修复

**Comment by MrWolong** [comment]
closed

**Pull request closed without merging**